src/routes/v2/collections.rs

```diff
@@ -0,0 +1,545 @@
+use crate::auth::checks::{filter_authorized_collections, is_authorized_collection};
+use crate::auth::get_user_from_headers;
+use crate::database;
+use crate::database::models::{collection_item, generate_collection_id, project_item};
+use crate::file_hosting::FileHost;
+use crate::models::collections::{Collection, CollectionStatus};
+use crate::models::ids::base62_impl::parse_base62;
+use crate::models::ids::{CollectionId, ProjectId};
+use crate::models::pats::Scopes;
+use crate::queue::session::AuthQueue;
+use crate::routes::ApiError;
+use crate::util::routes::read_from_payload;
+use crate::util::validate::validation_errors_to_string;
+use actix_web::web::Data;
+use actix_web::{delete, get, patch, post, web, HttpRequest, HttpResponse};
+use chrono::Utc;
+use serde::{Deserialize, Serialize};
+use sqlx::PgPool;
+use std::sync::Arc;
+use validator::Validate;
+
+use super::project_creation::CreateError;
+
+pub fn config(cfg: &mut web::ServiceConfig) {
+    cfg.service(collections_get);
+    cfg.service(collection_create);
+    cfg.service(
+        web::scope("collection")
+            .service(collection_get)
+            .service(collection_delete)
+            .service(collection_edit)
+            .service(collection_icon_edit)
+            .service(delete_collection_icon),
+    );
+}
+
+#[derive(Serialize, Deserialize, Validate, Clone)]
+pub struct CollectionCreateData {
+    #[validate(
+        length(min = 3, max = 64),
+        custom(function = "crate::util::validate::validate_name")
+    )]
+    #[serde(alias = "collection_name")]
+    /// The title or name of the project.
+    pub title: String,
+    #[validate(length(min = 3, max = 255))]
+    #[serde(alias = "collection_description")]
+    /// A short description of the collection.
+    pub description: String,
+    #[validate(length(max = 32))]
+    #[serde(default = "Vec::new")]
+    /// A list of initial projects to use with the created collection
+    pub initial_projects: Vec<String>,
```

Comment:
weird name for this field. maybe just projects? also should take in a Vec<ProjectId> (edit: oh wait, maybe not if you are allowing slugs as well, if so that's fine)